src/enzyme_ad/jax/Passes/PropagateConstantBound.cpp

```diff
@@ -129,6 +151,28 @@ struct PropagateConstantBoundsPass
                                       gridIdzOp.getOperation());
       });
     });
+
+    auto result = moduleOp->walk([&](enzymexla::KernelCallOp callOp) {
+      auto symbolName = callOp.getFn();
+      auto callee = symTable.lookup<FunctionOpInterface>(symbolName);
+      if (!callee)
+        return WalkResult::advance();
+      MLIRContext *ctx = callee->getContext();
```

Comment:
Something I realize here. this (and the above) is wrong. A callee could have multiple callers. You need to conservatively min the dereferencable size over all callers (and same for the range above, though that's a max)

Reply:
Mind taking another look? thanks!